command/plugin.go

```diff
@@ -0,0 +1,168 @@
+	"parallels-pvm":   new(parallelspvm.Builder),
+	"qemu":            new(qemu.Builder),
+	"virtualbox-iso":  new(virtualboxiso.Builder),
+	"virtualbox-ovf":  new(virtualboxovf.Builder),
+	"vmware-iso":      new(vmwareiso.Builder),
+	"vmware-vmx":      new(vmwarevmx.Builder),
+}
+
+var Provisioners = map[string]packer.Provisioner{
+	"ansible-local":     new(ansiblelocal.Provisioner),
+	"chef-client":       new(chefclient.Provisioner),
+	"chef-solo":         new(chefsolo.Provisioner),
+	"file":              new(fileprovisioner.Provisioner),
+	"powershell":        new(powershell.Provisioner),
+	"puppet-masterless": new(puppetmasterless.Provisioner),
+	"puppet-server":     new(puppetserver.Provisioner),
+	"salt-masterless":   new(saltmasterless.Provisioner),
+	"shell":             new(shell.Provisioner),
+	"shell-local":       new(shelllocal.Provisioner),
+	"windows-restart":   new(restart.Provisioner),
+	"windows-shell":     new(windowsshell.Provisioner),
+}
+
+var PostProcessors = map[string]packer.PostProcessor{
+	"artifice":      new(artifice.PostProcessor),
+	"atlas":         new(atlas.PostProcessor),
+	"compress":      new(compress.PostProcessor),
+	"docker-import": new(dockerimport.PostProcessor),
+	"docker-push":   new(dockerpush.PostProcessor),
+	"docker-save":   new(dockersave.PostProcessor),
+	"docker-tag":    new(dockertag.PostProcessor),
+	"vagrant":       new(vagrant.PostProcessor),
+	"vagrant-cloud": new(vagrantcloud.PostProcessor),
+	"vsphere":       new(vsphere.PostProcessor),
+}
+
+func (c *PluginCommand) Run(args []string) int {
+	// This is an internal call so we're not going to do much error checking.
+	// If there's a problem we'll usually just crash.
+	log.Printf("args: %#v", args)
+	if len(args) != 1 {
+		c.Ui.Error("Wrong number of args")
+		os.Exit(1)
+	}
+
+	// Plugin should be called like "packer-builder-amazon-ebs" so we'll take it
+	// apart.
+	parts := strings.Split(args[0], "-")
+	pluginType := parts[1]
+	pluginName := ""
+	// Post-processor is split so we'll so some magic here. We could use a
+	// regexp but this is simpler.
+	if pluginType == "post" {
+		pluginType = strings.Join(parts[1:2], "-")
+		pluginName = strings.Join(parts[3:], "-")
+	} else {
+		pluginName = strings.Join(parts[2:], "-")
+	}
+
+	server, err := plugin.Server()
+	if err != nil {
+		panic(err)
+	}
+
+	if pluginType == "builder" {
+		builder, found := Builders[pluginName]
+		if !found {
+			c.Ui.Error(fmt.Sprintf("Could not load builder: %s", pluginName))
+		}
+		server.RegisterBuilder(builder)
+	} else if pluginType == "provisioner" {
+		provisioner, found := Provisioners[pluginName]
+		if !found {
+			c.Ui.Error(fmt.Sprintf("Could not load provisioner: %s", pluginName))
+		}
+		server.RegisterProvisioner(provisioner)
+	} else if pluginType == "post-processor" {
+		postProcessor, found := PostProcessors[pluginName]
+		if !found {
+			c.Ui.Error(fmt.Sprintf("Could not load post-processor: %s", pluginName))
```

Comment:
Ahh good catch, I should probably exit or panic here. I use `ui.Error` in a few places to make debugging easier because packer suppresses the `log.Print` calls, but all of the failure scenarios will eventually crash.